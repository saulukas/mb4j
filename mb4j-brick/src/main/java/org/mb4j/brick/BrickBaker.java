package org.mb4j.brick;

public interface BrickBaker<P> {
  Brick bakeBrick(P params);
}
