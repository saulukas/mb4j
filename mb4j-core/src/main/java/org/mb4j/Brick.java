package org.mb4j;

import org.mb4j.renderer.RenderingScope;
import com.samskivert.mustache.Mustache.Lambda;
import com.samskivert.mustache.Template.Fragment;
import java.io.Writer;

public class Brick implements Lambda {
  @Override
  public void execute(Fragment frag, Writer out) {
    ((RenderingScope) out).renderIndented(this, frag);
  }
}
