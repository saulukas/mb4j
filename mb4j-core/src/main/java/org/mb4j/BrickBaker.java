package org.mb4j;

public interface BrickBaker<P> {
  Brick bakeBrick(P params);
}
