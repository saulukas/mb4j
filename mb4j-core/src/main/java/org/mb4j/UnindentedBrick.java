package org.mb4j;

import org.mb4j.renderer.RenderingScope;
import com.samskivert.mustache.Template.Fragment;
import java.io.Writer;

public class UnindentedBrick extends Brick {
  @Override
  public void execute(Fragment frag, Writer out) {
    ((RenderingScope) out).renderUnindented(this, frag);
  }
}
