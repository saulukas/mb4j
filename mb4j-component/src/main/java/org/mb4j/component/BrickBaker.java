package org.mb4j.component;

import org.mb4j.brick.MustacheBrick;

public interface BrickBaker {

    MustacheBrick bakeBrick(Request request);
}
