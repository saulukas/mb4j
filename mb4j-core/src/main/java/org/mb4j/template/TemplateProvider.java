package org.mb4j.template;

import org.mb4j.Brick;

public interface TemplateProvider {
  BrickTemplate templateFor(Class<? extends Brick> brickClass);
}
