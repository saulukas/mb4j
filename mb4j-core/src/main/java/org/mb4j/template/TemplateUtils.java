package org.mb4j.template;

import org.mb4j.Brick;

public class TemplateUtils {
  public static String typeStringOf(Class<? extends Brick> brickClass) {
    TemplateType annotation = brickClass.getAnnotation(TemplateType.class);
    return annotation == null ? TemplateType.DEFAULT : annotation.value();
  }

  public static String encodingStringOf(Class<? extends Brick> brickClass) {
    TemplateEncoding annotation = brickClass.getAnnotation(TemplateEncoding.class);
    return annotation == null ? TemplateEncoding.DEFAULT : annotation.value();
  }

  public static String outputEncodingStringOf(Class<? extends Brick> brickClass) {
    TemplateOutputEncoding annotation = brickClass.getAnnotation(TemplateOutputEncoding.class);
    return annotation == null ? TemplateOutputEncoding.DEFAULT : annotation.value();
  }
}
