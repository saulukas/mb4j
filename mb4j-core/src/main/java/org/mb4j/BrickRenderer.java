package org.mb4j;

import org.mb4j.renderer.RenderingScope;
import org.mb4j.template.TemplateProvider;
import java.io.Writer;

public class BrickRenderer {
  private final TemplateProvider templateProvider;

  public BrickRenderer(TemplateProvider templateProvider) {
    this.templateProvider = templateProvider;
  }

  public void render(Brick brick, Writer out) {
    RenderingScope scope = new RenderingScope(out, templateProvider);
    scope.render(brick);
  }
}
