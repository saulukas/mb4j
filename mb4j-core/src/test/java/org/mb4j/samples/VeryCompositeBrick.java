package org.mb4j.samples;

import org.mb4j.Brick;
import org.mb4j.BrickBaker;
import org.mb4j.template.TemplateType;

@TemplateType(".mustache")
public class VeryCompositeBrick extends Brick {
  final Brick moreComposite = new MoreCompositeBrick();
  final Brick composite = new CompositeBrick();
  final Brick simple = new SimpleBrick("<Hello>");
  final Brick listOfBricks = new ListOfBrickItems();
  final Brick listOfCustomItems = new ListOfCustomItems();
  String message;

  VeryCompositeBrick() {
  }

  public static class Baker implements BrickBaker<Baker.Params> {
    public VeryCompositeBrick bakeBrick(Params params) {
      VeryCompositeBrick brick = new VeryCompositeBrick();
      brick.message = params.message;
      return brick;
    }

    public static class Params {
      final String message;

      public Params(String message) {
        this.message = message;
      }
    }
  }
}
