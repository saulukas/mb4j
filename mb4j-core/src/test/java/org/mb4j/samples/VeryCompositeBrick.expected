
    Very composite brick wants to say Very Composite Hello :)
    =============================================================================
    .... 
    ....     More composite brick wants to say More Composite Hello :)
    ....     --------------------------------------------------------------------
    ....     composite1: 
    ....     
    ....         Composite brick wants to say Composite Hello 1 :)
    ....     
    ....         1. Just wanted to say First Hello :)
    ....         2. Just wanted to say Second Hello :)
    ....     
    ....     --------------------------------------------------------------------
    ....     composite2: 
    ....     -----------     Composite brick wants to say Composite Hello 2 :)
    ....     ----------- 
    ....     -----------     1. Just wanted to say First Hello :)
    ....     -----------     2. Just wanted to say Second Hello :)
    ....     ----------- 
    ....     --------------------------------------------------------------------
    ....     simple: Just wanted to say Simple Hello :)
    .... 
    =============================================================================
    .... 
    ....     Composite brick wants to say Composite Hello :)
    .... 
    ....     1. Just wanted to say First Hello :)
    ....     2. Just wanted to say Second Hello :)
    .... 
    =============================================================================
    .... Just wanted to say <Hello> :)
    =============================================================================
    .... 
    ....     List of bricks:
    .... 
    ....     Just wanted to say water2 :)
    ....     Just wanted to say bread2 :)
    ....     Just wanted to say onion2 :)
    .... 
    ....     Just wanted to say water2 :) <---> Just wanted to say bread2 :) <---> Just wanted to say onion2 :)
    .... 
    =============================================================================
    .... 
    ....     List of custom items:
    .... 
    ....     1. Just wanted to say water :)
    ....     2. Just wanted to say bread :)
    ....     3. Just wanted to say onion :)
    ....     
    =============================================================================
